src/mon/MonClient.cc

```diff
@@ -764,14 +772,19 @@ void MonClient::_renew_subs()
       sub_renew_sent = ceph_clock_now(cct);
 
     MMonSubscribe *m = new MMonSubscribe;
-    m->what = sub_have;
+    m->what = sub_new;
```

Comment:
This doesn't work, we need to combine sub_new and sub_sent whenever we _renew_subs(), or else we aren't including everything. We have to do that whenever we connect to a new monitor, and this is the path for that...
(See the osd_pg_creates subscription change in OSD::init for just one example of how this will otherwise break things.)